pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1510,18 +1523,90 @@ func getNetworkProviderConfig(ctx context.Context, configManager ConfigManager,
 		return nil, errorUtil.Wrap(err, "failed to unmarshal aws vpc create config")
 	}
 
-	if vpcCreateConfig.CidrBlock == nil {
-		return nil, errorUtil.New(fmt.Sprintf("CidrBlock in _network strategy tier %s can not be nil", tier))
+	// if the config map is found and the _network block contains an entry, that is returned for use in the network creation
+	if vpcCreateConfig.CidrBlock != nil && *vpcCreateConfig.CidrBlock != "" {
+		_, vpcCidr, err := net.ParseCIDR(*vpcCreateConfig.CidrBlock)
+		if err != nil {
+			return nil, errorUtil.Wrap(err, "failed to parse cidr block from _network strategy")
+		}
+		logger.Infof("found vpc cidr block %s in network strategy tier %s", vpcCidr.String(), tier)
+		return vpcCidr, nil
 	}
 
-	_, vpcCidr, err := net.ParseCIDR(*vpcCreateConfig.CidrBlock)
+	//if vpcCreateConfig.CidrBlock is nil or an empty string we can go ahead and set a sensible non overlapping default with a mask size of /26
+	defaultCIDR, err := n.getNonOverlappingDefaultCIDR(ctx)
 	if err != nil {
-		return nil, errorUtil.Wrap(err, "failed to parse cidr block from _network strategy")
+		return nil, errorUtil.Wrap(err, "failed to generate default CIDR")
 	}
-	cidrMask, _ := vpcCidr.Mask.Size()
 
-	logger.Infof("found vpc cidr block %s/%d in network strategy tier %s", vpcCidr.IP.String(), cidrMask, tier)
-	return vpcCidr, nil
+	// a default cidr is generated and updated to the config map, that is returned for use in the network creation
+	return defaultCIDR, nil
+}
+
+// getNonOverlappingDefaultCIDR returns a non overlapping cidr block based on the OpenShift vpc cidr block
+// the default mask is /26
+// for other masks the user is required to provide their own via config
+func (n *NetworkProvider) getNonOverlappingDefaultCIDR(ctx context.Context) (*net.IPNet, error) {
+	clusterVpc, err := getClusterVpc(ctx, n.Client, n.Ec2Api, n.Logger)
+	if err != nil {
+		errMsg := "failed to get cluster vpc for cidr block"
```

Comment:
just a question, is there a reason why we are not keeping it the same as other .Wrap logging?

Reply:
Its left over from a copy and paste, in other parts of cro, we return a status along with the error. So its easier to declare the `errMsg` and return it in both the status and errorUtil.Wrap. 

Example : https://github.com/integr8ly/cloud-resource-operator/blob/master/pkg/providers/aws/provider_postgres.go#L145

Reply:
I have to make another change so I'll clean this up. 👍